File: app/src/main/java/abdullah/elamien/bloodbank/utils/ConfigHelper.java
package abdullah.elamien.bloodbank.utils;

import android.content.res.Resources;
import android.util.DisplayMetrics;

import java.util.Locale;

public class ConfigHelper {
    public static void setLanguage(Resources resources, String languageCode) {
        DisplayMetrics displayMetrics = resources.getDisplayMetrics();
        android.content.res.Configuration configuration = resources.getConfiguration();
        configuration.setLocale(new Locale(languageCode.toLowerCase()));
        resources.updateConfiguration(configuration, displayMetrics);
    }
}

File: app/src/main/java/abdullah/elamien/bloodbank/utils/ConfigHelper.kt
package abdullah.elamien.bloodbank.utils

import android.content.res.Resources
import java.util.*

object ConfigHelper {
    fun setLanguage(resources: Resources, languageCode: String) {
        val displayMetrics = resources.displayMetrics
        val configuration = resources.configuration
        configuration.setLocale(Locale(languageCode.toLowerCase()))
        resources.updateConfiguration(configuration, displayMetrics)
    }
}
